### DISCUSSION TITLE
Issue in Speaker Diarization with ASR

### DISCUSSION THREAD

**Discussion by viralrupapara36** [NVIDIA-NeMo/NeMo#5145] [Q&A] [answered]
In demo audio file diarization with ASR works well but while testing other audio files normal diarization working good but while running same audio files with ASR it cannot detect another speaker, It only detect one speaker.

Means, normal diarization and diarization with asr producing different diarization results.

**Comment by tango4j** [comment, marked as answer]
You should be using the exact same YAML files for the inference. Otherwise, the two diarization results will show difference. Also, please be sure to use the latest NeMo toolkit version.

Also, If the estimated number of speakers is not so accurate, be sure to turn of `asr_based_vad` off in the inference config.

```
    diarizer.asr.parameters.asr_based_vad=False
```

We recommend the following config for the best possible outcome for diarization and ASR.
```
export CUDA_VISIBLE_DEVICES=0
MANIFEST_FILE='/data/sample.json'
cd $NeMo/examples/speaker_tasks/diarization
 
python ./clustering_diarizer/offline_diar_with_asr_infer.py \
        --config-path='./../conf/inference' --config-name='diar_infer_telephonic.yaml' \
    diarizer.manifest_filepath=$MANIFEST_FILE \
    diarizer.out_dir='/data/sample_readAI' \
    diarizer.speaker_embeddings.model_path='titanet_large' \
   diarizer.clustering.parameters.oracle_num_speakers=False \
    diarizer.oracle_vad=False \
    diarizer.vad.model_path='vad_multilingual_marblenet' \
    diarizer.asr.model_path='stt_en_conformer_ctc_large' \
    diarizer.asr.parameters.asr_based_vad=False \
    diarizer.asr.parameters.asr_based_vad_threshold=1 \
    diarizer.msdd_model.model_path=None \
    diarizer.vad.parameters.smoothing=median \
    diarizer.vad.parameters.pad_onset=0.1 \
    num_workers=1
```
